app/src/main/java/com/infomaniak/drive/data/models/File.kt

```diff
@@ -422,5 +432,13 @@ open class File(
             val mediaFolder = context.externalMediaDirs?.firstOrNull() ?: context.filesDir
             return java.io.File(mediaFolder, context.getString(R.string.EXPOSED_OFFLINE_DIR))
         }
+
+        fun File.getColor(context: Context): String {
+            return _color ?: String.format("#%06x", ContextCompat.getColor(context, R.color.secondaryText) and 0xffffff)
+        }
+
+        fun File.setColor(color: String) {
+            _color = color
+        }
```

Comment:
Don't do extensions for this

Take inspiration from this: https://github.com/Infomaniak/android-kDrive/blob/master/app/src/main/java/com/infomaniak/drive/data/models/drive/Drive.kt#L75